launcher/ui/pages/instance/ServersPage.cpp

```diff
@@ -112,11 +125,42 @@ struct Server {
     bool m_checked = false;
     bool m_up = false;
     QString m_motd;  // https://mctools.org/motd-creator
-    int m_ping = 0;
-    int m_currentPlayers = 0;
+    std::optional<int> m_currentPlayers; // nullopt if not calculated/calculating
     int m_maxPlayers = 0;
 };
 
+class ServerPingTask : public Task {
+    Q_OBJECT
+  public:
+    explicit ServerPingTask(QObject* parent, Server &server) : Task(parent), m_server(server) {}
+    ~ServerPingTask() override = default;
+
+
+  protected:
+    virtual void executeTask() override {
+        qDebug() << "Querying status of " << m_server.m_address;
+        auto [domain, port] = m_server.splitAddress();
+        McResolver *resolver = new McResolver(nullptr, domain, port);
```

Comment:
This is a leak: either use smart pointers as class members or connect the fail signal to deleteLater

Reply:
Shouldn't adding a McResolver instance as a class member (without using a smart pointer) suffice ?

Btw, could you explain to me what a Task is, and if I should even use one here ? Right now I don't see its utility at all, I could call my code directly